tests/test_project.py

```diff
@@ -562,6 +563,70 @@ def test_init_local_missing_west_yml_failure(repos_tmpdir):
         cmd(f'init -l "{zephyr_install_dir}"')
 
 
+def test_init_local_with_manifest_filename(repos_tmpdir):
+    # Test 'west init --mf -l' on a local repo
+
+    manifest = repos_tmpdir / 'repos' / 'zephyr'
+    workspace = repos_tmpdir / 'workspace'
+    zephyr_install_dir = workspace / 'zephyr'
+
+    # Do a local clone of manifest repo
+    clone(str(manifest), str(zephyr_install_dir))
+    os.rename(str(zephyr_install_dir / 'west.yml'),
+              str(zephyr_install_dir / 'project.yml'))
+
+    # fails because west.yml is missing
+    with pytest.raises(subprocess.CalledProcessError):
+        cmd(f'init -l "{zephyr_install_dir}"')
+
+    with open(zephyr_install_dir / 'west.yml', 'w') as f:
+        f.write('[')
+
+    # fails because of the syntax error
+    cwd = os.getcwd()
+
+    with pytest.raises(subprocess.CalledProcessError):
+        cmd(f'init -l "{zephyr_install_dir}"')
+        # init with a local manifest doesn't parse the file, so let's access it
+        workspace.chdir()
+        cmd('list')
```

Comment:
This looks like dead code to me. The line `cmd(f'init -l "{zephyr_install_dir}"')` should raise the exception. Am I missing something?

Reply:
I don't know if this is intended behavior or not, but as the comment above says, it looks like when using a local manifest west doesn't seem to parse that file at all because even syntax errors result in successful initialization.

Reply:
Yep, this is intended behavior. We only need to parse the manifest when initializing from the network, because `self: path: foo` in the manifest file means "put this repository in directory `foo` in the initialized workspace when you're done". That's what this comment is hinting at: https://github.com/zephyrproject-rtos/west/blob/master/src/west/app/project.py#L267

By contrast, `west init -l foo` means "`foo` is already exactly where I want it; initialize a workspace around it". There's no need to parse the manifest at init time in that case.

Reply:
@M1cha any plans to update this? I think I've answered outstanding questions. Please let me know if I missed anything and thanks again for your work.

Reply:
I got sick but am feeling better now. I'll update the PR today :)